src/server/node_manager.c

```diff
@@ -3663,14 +3761,18 @@ update2_to_vnode(vnal_t *pvnal, int new, mominfo_t *pmom, int *madenew, int from
 				}
 			}
 		}
-	} else {
-		/* cross link the vnode and its Mom */
+	} else if (from_hook != 2) {
```

Comment:
Do you want an "else if" here? Shouldn't this check happen regardless of whether the "if" condition were true?